jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -240,13 +240,6 @@ MethodHandle cloneWithNewType(MethodType newType) {
 	// Until the JIT can synthesize calls to virtual methods, we must synthesize calls to these static ones instead
 	/*[ENDIF]*/
 	private static MethodHandle asType(MethodHandle mh, MethodType newType) {
-		/*
```

Comment:
Why is this being deleted?

Reply:
That code was added because we wanted to inine `asType` and let VP fold the comparison to a goto. We don't need it now because VP recognize and fold the outer method.

Reply:
Is the VM ever running this code and having that if was saving us from creating new handles or was it only ever called from the JIT?

Reply:
No, having that won't save us from creating new handles. The intention for this code was to aid the optimization in the JIT. But now the JIT can do the same without the early return. The other MethodHandle.asType calls to MethodType.equals which does the same check.